src/cooldown.rs

```diff
@@ -3,18 +3,22 @@
 use crate::serenity_prelude as serenity;
 // I usually don't really do imports, but these are very convenient
 use std::collections::HashMap;
+use std::fmt::Debug;
+use std::sync::Arc;
 use std::time::{Duration, Instant};
 
 /// Subset of [`crate::Context`] so that [`Cooldowns`] can be used without requiring a full [Context](`crate::Context`)
 /// (ie from within an `event_handler`)
-#[derive(Default, Clone, PartialEq, Eq, Debug, Hash)]
-pub struct CooldownContext {
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct CooldownContext<'a, U> {
```

Comment:
Is there anything else we should add to this struct? This struct was factored out of `poise::Context` to be the minimal set of data needed to calculate cooldowns so it should be kept small and probably shouldn't have `non_exhaustive` to enable its use in the widest range of use-cases.

Potential things available within an `event_handler that we could add:
| Data| Comments|
|--------|--------|
| The `Event` itself | Probably not directly relevant to Cooldowns |
| The `FrameworkContext`| Only relevant fields are the user data and maybe the bot_id |
| The `serentiy::Context` | Probably not relevant for computing the cooldowns |